plugins/ptp_operator/event/event.go

```diff
@@ -135,22 +176,24 @@ func (p *PTPEventState) UpdateCurrentEventState(c ClockState) ptp.SyncState {
 				"node": clockState.NodeName, "iface": alias}).Set(float64(v))
 		}
 		clockState.Metric = metrics
-		p.DependsOn[c.Process] = clockState
+		p.DependsOn[c.Process] = []*ClockState{clockState}
 	}
 
 	// if all locked then its locked
 	// if anyone HOLDOVER then holdover
 	// if  anyone FREERUN then freerun
 	var currentState ptp.SyncState
 	for _, v := range p.DependsOn {
-		if v.State != ptp.LOCKED {
-			if v.State == ptp.HOLDOVER {
-				currentState = v.State
-				break
+		for _, dd := range v {
+			if dd.State != ptp.LOCKED {
+				if dd.State == ptp.HOLDOVER {
+					currentState = dd.State
+					break
+				}
+				currentState = dd.State
```

Comment:
if v[0]=FREERUN and v[1]=HOLDOVER,  currentState will be set to HOLDOVER, which is not correct